### PR TITLE
[semi-modular][ready]YEE HAW! More cowboy clothes!

### PR DESCRIPTION
**Pull request by thestubborn**
## About The Pull Request

title, made by Hay#7679 whom gave me the honour to add them
image
image

## Why It's Good For The Game
https://www.youtube.com/watch?v=OM0ilkux9_M
clothing good

## Changelog
:cl: Big 'Stiff' Stubborn, head of the Hat Gang. Ragin' 'Lilliputian' Lily, Deer 'Dear' Lea and Crazy 'Zanzibarrian' ZephyrTFA.
expansion: YEE HAW! More cowboy stuff! Hats, uniforms, suit, a poncho, a belt!
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/customization/modules/clothing/neck/_neck.dm [hunk 1]
```diff
@@ -5,3 +5,38 @@
 	icon_state = "eldritch_tie"
 	desc = "The necktie is adorned with a garish pattern. It's disturbingly vivid. Somehow you feel as if it would be wrong to ever take it off. It's your friend now. You will betray it if you change it for some boring scarf."
 
+var/togglename = null
```
Comment (FlamingLily) [inline modular_skyrat/modules/customization/modules/clothing/neck/_neck.dm]:
```suggestion
var/togglename
```
vars are null by default

File: modular_skyrat/modules/customization/modules/clothing/neck/_neck.dm [hunk 2]
```diff
@@ -5,3 +5,38 @@
...
+var/src.necktoggled
```
Comment (FlamingLily) [inline modular_skyrat/modules/customization/modules/clothing/neck/_neck.dm]:
```suggestion
var/necktoggled
```
That's not really needed

File: modular_skyrat/modules/customization/modules/clothing/neck/_neck.dm [hunk 3]
```diff
@@ -5,3 +5,38 @@
...
+	neck_toggle()
+
+/obj/item/clothing/neck/toggle/proc/neck_toggle()
+	set src in usr
+
+	if(!can_use(usr))
+		return 0
+
+	to_chat(usr, span_notice("You toggle [src]'s [togglename]."))
+	if(src.necktoggled)
```
Comment (FlamingLily) [inline modular_skyrat/modules/customization/modules/clothing/neck/_neck.dm]:
```suggestion
	if(necktoggled)
```
get rid of all src.necktoggled and replace with necktoggled

File: modular_skyrat/modules/customization/modules/clothing/neck/_neck.dm [hunk 4]
```diff
@@ -5,3 +5,38 @@
...
+		src.icon_state = "[initial(icon_state)]"
+		src.necktoggled = FALSE
```
Comment (FlamingLily) [inline modular_skyrat/modules/customization/modules/clothing/neck/_neck.dm]:
```suggestion
		necktoggled = FALSE
```

File: modular_skyrat/modules/customization/modules/clothing/neck/_neck.dm [hunk 5]
```diff
@@ -5,3 +5,38 @@
...
+	else if(!src.necktoggled)
```
Comment (FlamingLily) [inline modular_skyrat/modules/customization/modules/clothing/neck/_neck.dm]:
```suggestion
	else if(!necktoggled)
```

File: modular_skyrat/modules/customization/modules/clothing/neck/_neck.dm [hunk 6]
```diff
@@ -5,3 +5,38 @@
...
+		src.icon_state = "[initial(icon_state)]_t"
+		src.necktoggled = TRUE
```
Comment (FlamingLily) [inline modular_skyrat/modules/customization/modules/clothing/neck/_neck.dm]:
```suggestion
		necktoggled = TRUE
```

File: modular_skyrat/modules/customization/modules/clothing/neck/cloaks.dm [hunk 7]
```diff
@@ -12,3 +12,14 @@
...
+
+/obj/item/clothing/neck/toggle/cowboylea
+	name = "green cowboy ponch"
+	desc = "A sand covered cloak, there seems to be a small deer head with antlers embroidered inside."
+	icon = 'modular_skyrat/master_files/icons/obj/clothing/neck.dmi'
+	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/neck.dmi'
+	icon_state = "cowboy_poncho"
+	mutant_variants = NONE
+	heat_protection = CHEST
+	blood_overlay_type = "suit"
```
Comment (FlamingLily) [inline modular_skyrat/modules/customization/modules/clothing/neck/cloaks.dm]: figure out what blood_overlay_type does, if anything.

**Review by FlamingLily — COMMENTED** [review]
So, because just telling you what to do doesn't usually help, I'm going to explain why

when defining a variable with `var/variable` that variable is a direct child of the source. Or, "src," You don't need to call src.variable, because just calling variable will do that.

File: modular_skyrat/modules/customization/modules/clothing/head/head.dm [hunk 8]
```diff
@@ -397,3 +398,30 @@
 	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/head.dmi'
 	icon_state = "enhead"
 	mutant_variants = NONE
+
+/obj/item/clothing/head/cowboyhat/sherrif
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/clothing/head/head.dm:402]:
```suggestion
/obj/item/clothing/head/cowboyhat/sheriff
```

File: modular_skyrat/modules/customization/modules/clothing/under/security.dm [hunk 9]
```diff
@@ -54,6 +54,41 @@
 	mutant_variants = NONE
 	heat_protection = CHEST|ARMS
 
+/obj/item/clothing/under/rank/security/detective/cowboy
+	name = "blond cowboy uniform"
+	desc = "A blue shirt with some cool Cowboy socks. You dig."
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/clothing/under/security.dm:59]:
```suggestion
	desc = "A blue shirt with some cool cowboy socks. You dig."
```

File: modular_skyrat/modules/customization/modules/clothing/under/security.dm [hunk 10]
```diff
@@ -54,6 +54,41 @@
...
+	icon_state = "cowboy_uniform"
+	mutant_variants = NONE
+	can_adjust = FALSE
+
+/obj/item/clothing/under/rank/security/detective/cowboy/armorless
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 0, ACID = 0)
+
+/obj/item/clothing/suit/cowboyvest
+	name = "blonde cowboy vest"
+	desc = "A white cream vest lined with...fur, of all things, for desert weather. There's a small Deer head logo sewn in to the vest."
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/clothing/under/security.dm:71]:
```suggestion
	desc = "A white cream vest lined with... fur, of all things, for desert weather. There's a small deer head logo sewn into the vest."
```

File: modular_skyrat/modules/customization/modules/clothing/under/security.dm [hunk 11]
```diff
@@ -54,6 +54,41 @@
...
+	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/suit.dmi'
+	icon_state = "cowboy_vest"
+	body_parts_covered = CHEST|ARMS
+	cold_protection = CHEST|ARMS
+	mutant_variants = NONE
+	heat_protection = CHEST|ARMS
+
+/obj/item/clothing/suit/det_suit/cowboyvest
+	name = "blonde cowboy vest"
+	desc = "A white cream vest lined with...fur, of all things, for desert weather. There's a small Deer head logo sewn in to the vest."
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/clothing/under/security.dm:82]: Not sure if all this stuff needs to be duplicated, but

```suggestion
	desc = "A white cream vest lined with... fur, of all things, for desert weather. There's a small deer head logo sewn into the vest."
```

File: modular_skyrat/modules/customization/modules/clothing/head/head.dm [hunk 12]
```diff
@@ -397,3 +398,30 @@
...
+	icon = 'modular_skyrat/master_files/icons/obj/clothing/hats.dmi'
+	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/head.dmi'
+	icon_state = "sheriff_hat"
+	mutant_variants = NONE
+	desc = "A dark hat from the cold wastes of the Frosthill mountains. So it was done, all according to the law. There's a small set of antlers embroidered on the inside."
+	cold_protection = HEAD
+	min_cold_protection_temperature = FIRE_HELM_MIN_TEMP_PROTECT
+	flags_inv = HIDEHAIR
+
+/obj/item/clothing/head/cowboyhat/sherrif/alt
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/clothing/head/head.dm:413]:
```suggestion
/obj/item/clothing/head/cowboyhat/sheriff/alt
```

File: modular_skyrat/modules/customization/modules/clothing/head/head.dm [hunk 13]
```diff
@@ -112,7 +112,7 @@
 
 /obj/item/clothing/head/cowboyhat/widesec
 	name = "wide brimmed security cowboy hat"
-	desc = "A bandit turned Sherriff, his enforcement is brutal but effective, if out of fear or respect, not many bodies hang high. A peaceful land, a quiet people."
+	desc = "A bandit turned Sheriff, his enforcement is brutal but effective, if out of fear or respect, not many bodies hang high. A peaceful land, a quiet people."
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/clothing/head/head.dm:115]:
```suggestion
	desc = "A bandit turned sheriff, his enforcement is brutal but effective, if out of fear or respect, not many bodies hang high. A peaceful land, a quiet people."
```

File: modular_skyrat/modules/customization/modules/client/loadout/neck.dm [hunk 14]
```diff
@@ -39,6 +39,10 @@
 	name = "Holocollar"
 	path = /obj/item/clothing/neck/human_petcollar/locked/holocollar
 
+/datum/loadout_item/neck/poncxhocowbot
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/client/loadout/neck.dm:42]:
```suggestion
/datum/loadout_item/neck/ponchocowboy
```

I assume this isn't intentional

File: code/modules/vending/clothesmate.dm [hunk 15]
```diff
@@ -159,6 +159,9 @@
 					/obj/item/clothing/head/hijab = 3,
 					/obj/item/clothing/head/turb = 3,
 					/obj/item/clothing/head/keffiyeh = 3,
+					/obj/item/clothing/head/cowboyhat/sherrif = 3,
+					/obj/item/clothing/head/cowboyhat/sherrif/alt = 3,
```
Comment (tf-4) [inline code/modules/vending/clothesmate.dm:163]:
```suggestion
					/obj/item/clothing/head/cowboyhat/sheriff = 3,
					/obj/item/clothing/head/cowboyhat/sheriff/alt = 3,
```

File: code/modules/vending/autodrobe.dm [hunk 16]
```diff
@@ -156,6 +156,9 @@
 					/obj/item/clothing/suit/dutchjacketsr = 1,
 					/obj/item/clothing/mask/gas/polyclown = 1,
 					/obj/item/clothing/mask/gas/clownbald = 1,
+					/obj/item/clothing/head/cowboyhat/sherrif = 1,
+					/obj/item/clothing/head/cowboyhat/sherrif/alt = 1,
```
Comment (tf-4) [inline code/modules/vending/autodrobe.dm:160]:
```suggestion
					/obj/item/clothing/head/cowboyhat/sheriff = 1,
					/obj/item/clothing/head/cowboyhat/sheriff/alt = 1,
```

File: modular_skyrat/modules/customization/modules/client/loadout/head.dm [hunk 17]
```diff
@@ -158,6 +158,18 @@
 	name = "Cowboy Hat, Pink"
 	path = /obj/item/clothing/head/cowboyhat/pink
 
+/datum/loadout_item/head/cowboyhat/winter
+	name = "Winter Cowboy Hat"
+	path = /obj/item/clothing/head/cowboyhat/sherrif
+
+/datum/loadout_item/head/cowboyhat/sherrif
+	name = "Sherrif Hat"
+	path = /obj/item/clothing/head/cowboyhat/sherrif/alt
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/client/loadout/head.dm:167]:
```suggestion
	path = /obj/item/clothing/head/cowboyhat/sheriff

/datum/loadout_item/head/cowboyhat/sheriff
	name = "Sheriff Hat"
	path = /obj/item/clothing/head/cowboyhat/sheriff/alt
```

**Review by tf-4 — COMMENTED** [review]
minor spelling and grammar clean-up

**Pull request merged**